structopt-derive/src/parse.rs

```diff
@@ -0,0 +1,158 @@
+use syn::parse::{Parse, ParseStream};
+use syn::punctuated::Punctuated;
+use syn::{self, Attribute, Expr, Ident, LitStr};
+
+pub struct StructOptAttributes {
+    pub paren_token: syn::token::Paren,
+    pub attrs: Punctuated<StructOptAttr, Token![,]>,
+}
+
+impl Parse for StructOptAttributes {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let content;
+        Ok(StructOptAttributes {
+            paren_token: parenthesized!(content in input),
```

Comment:
From where is this macro? syn?

Reply:
Yes, it's from `syn`: https://docs.rs/syn/0.15.35/syn/macro.parenthesized.html